src/execution/base.ts

```diff
@@ -102,4 +102,37 @@ export abstract class BaseExecutionWithCommand<
    * needed to run at all.
    */
   readonly servicesNotNeeded = this._servicesNotNeeded.promise;
+
+  /**
+   * Resolves when any of the services this script depends on have terminated
+   * (see {@link ServiceScriptExecution.terminated} for exact definiton).
+   */
+  readonly anyServiceTerminated = Promise.race(
+    this._config.services.map(
+      (service) => this._executor.getExecution(service).terminated
+    )
+  );
+
+  /**
+   * Ensure that all of the services this script depends on are running.
+   */
+  protected async _startServices(): Promise<Result<void, Failure[]>> {
+    if (this._config.services.length > 0) {
+      const results = await Promise.all(
```

Comment:
Is `Promise.all()` ok here (over `Promise.allSettled()`) because you have your own error results that resolved to, not rejected?

Reply:
Yeah we throw only as a panic for an internal error, so in general we don't need to catch exceptions.